### ISSUE TITLE
Nested MLC output doesn't resolve a value on NodeJS SDK

### ISSUE THREAD

**Issue by jaxxstorm** [pulumi/pulumi#14878]
### What happened?

I have an MLC: https://github.com/lbrlabs/pulumi-lbrlabs-eks

I have an example program in python and YAML that works great. The same program in TypeScript fails to attach a nodegroup due to what appears to be an output that doesn't resolve

### Example

```
import * as pulumi from "@pulumi/pulumi";
import * as awsx from "@pulumi/awsx";
import * as kubernetes from "@pulumi/kubernetes";
import * as lbrlabs_eks from "@lbrlabs/pulumi-eks";

const vpc = new awsx.ec2.Vpc("vpc", {
  cidrBlock: "172.16.0.0/16",
  subnetStrategy: awsx.ec2.SubnetAllocationStrategy.Auto,
  subnetSpecs: [
    {
      type: awsx.ec2.SubnetType.Public,
      cidrMask:20,
      tags: {
        "kubernetes.io/role/elb": "1",
      },
    },
    {
      type: awsx.ec2.SubnetType.Private,
      cidrMask: 19,
      tags: {
        "kubernetes.io/role/internal-elb": "1",
      },
    },
  ],
});
const cluster = new lbrlabs_eks.Cluster("cluster", {
  clusterSubnetIds: vpc.privateSubnetIds,
  letsEncryptEmail: "mail@lbrlabs.com",
  systemNodeSubnetIds: vpc.publicSubnetIds,
  systemNodeDesiredCount: 4,
});

export const clusterName = cluster.controlPlane.name;

const workloadNodes = new lbrlabs_eks.AttachedNodeGroup("workload", {
  clusterName: cluster.controlPlane.name,
  subnetIds: vpc.privateSubnetIds,
  scalingConfig: {
    desiredSize: 3,
    maxSize: 10,
    minSize: 1,
  },
});
```

Throws the error:

```
error: aws:eks/nodeGroup:NodeGroup resource 'workload-nodes' has a problem: "cluster_name" length must be between 1-100 characters: "". Examine values at 'workload-nodes.clusterName'.
    error: aws:eks/nodeGroup:NodeGroup resource 'workload-nodes' has a problem: "cluster_name" doesn't comply with restrictions ("^[0-9A-Za-z][0-9A-Za-z_-]+$"): "". Examine values at 'workload-nodes.clusterName'.
```

Note, the same example program in python works fine.

### Output of `pulumi about`

```
CLI          
Version      3.97.0
Go Version   go1.21.5
Go Compiler  gc

Plugins
NAME         VERSION
aws          6.14.0
awsx         2.3.0
docker       4.5.0
docker       3.6.1
kubernetes   3.29.1
lbrlabs-eks  0.2.3+dirty
nodejs       unknown

Host     
OS       darwin
Version  14.1.2
Arch     arm64

This project is written in nodejs: executable='/opt/homebrew/bin/node' version='v21.1.0'

Current Stack: lbrlabs58/lbrlabs-eks-ts/dev

TYPE                                                 URN
pulumi:pulumi:Stack                                  urn:pulumi:dev::lbrlabs-eks-ts::pulumi:pulumi:Stack::lbrlabs-eks-ts-dev
pulumi:providers:awsx                                urn:pulumi:dev::lbrlabs-eks-ts::pulumi:providers:awsx::default_2_3_0
pulumi:providers:aws                                 urn:pulumi:dev::lbrlabs-eks-ts::pulumi:providers:aws::default_6_9_0
awsx:ec2:Vpc                                         urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc::vpc
aws:ec2/vpc:Vpc                                      urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc::vpc
aws:ec2/subnet:Subnet                                urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::vpc-private-1
aws:ec2/internetGateway:InternetGateway              urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/internetGateway:InternetGateway::vpc
aws:ec2/subnet:Subnet                                urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::vpc-private-2
aws:ec2/subnet:Subnet                                urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::vpc-private-3
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::vpc-private-1
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::vpc-private-2
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::vpc-private-1
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::vpc-private-3
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::vpc-private-2
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::vpc-private-3
aws:ec2/subnet:Subnet                                urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::vpc-public-2
aws:ec2/subnet:Subnet                                urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::vpc-public-3
aws:ec2/subnet:Subnet                                urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::vpc-public-1
aws:ec2/eip:Eip                                      urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/eip:Eip::vpc-2
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::vpc-public-2
aws:ec2/eip:Eip                                      urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/eip:Eip::vpc-3
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::vpc-public-3
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::vpc-public-1
aws:ec2/eip:Eip                                      urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/eip:Eip::vpc-1
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::vpc-public-2
aws:ec2/route:Route                                  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::vpc-public-2
aws:ec2/route:Route                                  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::vpc-public-3
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::vpc-public-3
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::vpc-public-1
aws:ec2/route:Route                                  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::vpc-public-1
aws:ec2/natGateway:NatGateway                        urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/natGateway:NatGateway::vpc-2
aws:ec2/route:Route                                  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::vpc-private-2
aws:ec2/natGateway:NatGateway                        urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/natGateway:NatGateway::vpc-1
aws:ec2/route:Route                                  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::vpc-private-1
aws:ec2/natGateway:NatGateway                        urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/natGateway:NatGateway::vpc-3
aws:ec2/route:Route                                  urn:pulumi:dev::lbrlabs-eks-ts::awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::vpc-private-3
pulumi:providers:pulumi                              urn:pulumi:dev::lbrlabs-eks-ts::pulumi:providers:pulumi::default
pulumi:providers:lbrlabs-eks                         urn:pulumi:dev::lbrlabs-eks-ts::pulumi:providers:lbrlabs-eks::default_0_2_3_dirty_github_/api.github.com/lbrlabs
pulumi:providers:aws                                 urn:pulumi:dev::lbrlabs-eks-ts::pulumi:providers:aws::default
lbrlabs-eks:index:Cluster                            urn:pulumi:dev::lbrlabs-eks-ts::lbrlabs-eks:index:Cluster::cluster
aws:iam/role:Role                                    urn:pulumi:dev::lbrlabs-eks-ts::lbrlabs-eks:index:Cluster$aws:iam/role:Role::cluster-cluster-role
aws:iam/policy:Policy                                urn:pulumi:dev::lbrlabs-eks-ts::lbrlabs-eks:index:Cluster$aws:iam/role:Role$aws:iam/policy:Policy::cluster-cert-manager-policy
aws:iam/rolePolicyAttachment:RolePolicyAttachment    urn:pulumi:dev::lbrlabs-eks-ts::lbrlabs-eks:index:Cluster$aws:iam/role:Role$aws:iam/rolePolicyAttachment:RolePolicyAttachment::cluster-cluster-policy
aws:iam/policy:Policy                                urn:pulumi:dev::lbrlabs-eks-ts::lbrlabs-eks:index:Cluster$aws:iam/role:Role$aws:iam/policy:Policy::cluster-external-dns-policy
aws:iam/rolePolicyAttachment:RolePolicyAttachment    urn:pulumi:dev::lbrlabs-eks-ts::lbrlabs-eks:index:Cluster$aws:iam/role:Role$aws:iam/rolePolicyAttachment:RolePolicyAttachment::cluster-cluster-policy-vpc
aws:kms/key:Key                                      urn:pulumi:dev::lbrlabs-eks-ts::lbrlabs-eks:index:Cluster$aws:kms/key:Key::cluster-cluster-kms-key
aws:iam/policy:Policy                                urn:pulumi:dev::lbrlabs-eks-ts::lbrlabs-eks:index:Cluster$aws:iam/role:Role$aws:iam/policy:Policy::cluster-cluster-kms-policy
aws:iam/rolePolicyAttachment:RolePolicyAttachment    urn:pulumi:dev::lbrlabs-eks-ts::lbrlabs-eks:index:Cluster$aws:iam/role:Role$aws:iam/rolePolicyAttachment:RolePolicyAttachment::cluster-cluster-policy-kms


Found no pending operations associated with dev

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/lbrlabs58
User           lbrlabs58
Organizations  lbrlabs58, lbrlabs
Token type     personal

Dependencies:
NAME                 VERSION
@pulumi/kubernetes   3.29.1
@pulumi/pulumi       3.72.2
@types/node          14.18.51
typescript           4.9.5
@lbrlabs/pulumi-eks  0.2.3+dirty
@pulumi/aws          6.14.0
@pulumi/awsx         2.3.0

Pulumi locates its logs in /var/folders/rk/2c715ngd14jc4c7d3659m7pm0000gn/T/ by default
```

### Additional context

If I output the cluster name, it's empty. If I export the whole cluster object, it only outputs the cluster name

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by jaxxstorm** [comment]
Did more investigation here. Here's a python example:

```
import pulumi
import lbrlabs_pulumi_eks as lbrlabs_eks
import pulumi_aws as aws
import pulumi_awsx as awsx
import pulumi_kubernetes as kubernetes

vpc = awsx.ec2.Vpc(
    "vpc",
    cidr_block="172.16.0.0/16",
    subnet_strategy=awsx.ec2.SubnetAllocationStrategy.AUTO,
    subnet_specs=[
        awsx.ec2.SubnetSpecArgs(
            type=awsx.ec2.SubnetType.PUBLIC,
            cidr_mask=20,
            tags={
                "kubernetes.io/role/elb": "1",
            },
        ),
        awsx.ec2.SubnetSpecArgs(
            type=awsx.ec2.SubnetType.PRIVATE,
            cidr_mask=19,
            tags={
                "kubernetes.io/role/internal-elb": "1",
            },
        ),
    ],
)
cluster = lbrlabs_eks.Cluster(
    "cluster",
    cluster_subnet_ids=vpc.public_subnet_ids,
    system_node_subnet_ids=vpc.private_subnet_ids,
    system_node_instance_types=["t3.large"],
    system_node_desired_count=4,
    lets_encrypt_email="mail@lbrlabs.com",
)

pulumi.export("cluster_name", cluster.control_plane.name)
```

This outputs the cluster name. Here's a nodejs example:

```
const vpc = new awsx.ec2.Vpc("vpc", {
  cidrBlock: "172.16.0.0/16",
  subnetStrategy: awsx.ec2.SubnetAllocationStrategy.Auto,
  subnetSpecs: [
    {
      type: awsx.ec2.SubnetType.Public,
      cidrMask:20,
      tags: {
        "kubernetes.io/role/elb": "1",
      },
    },
    {
      type: awsx.ec2.SubnetType.Private,
      cidrMask: 19,
      tags: {
        "kubernetes.io/role/internal-elb": "1",
      },
    },
  ],
});
const cluster = new lbrlabs_eks.Cluster("cluster", {
  clusterSubnetIds: vpc.privateSubnetIds,
  letsEncryptEmail: "mail@lbrlabs.com",
  systemNodeSubnetIds: vpc.publicSubnetIds,
  systemNodeDesiredCount: 4,
});

export const clusterName = cluster.controlPlane.name;
```

This throws:

```
warning: Undefined value (clusterName) will not show as a stack output.
```

**Comment by jaxxstorm** [comment]
Looks to be related to #13802

**Comment by justinvp** [comment]
This looks like a duplicate of https://github.com/pulumi/pulumi/issues/13223. It's not related to https://github.com/pulumi/pulumi/issues/13802

The Node.js example from https://github.com/pulumi/pulumi/issues/14878#issuecomment-1861841803 works for me without warnings or errors using the published `"@lbrlabs/pulumi-eks": "0.3.0"` and associated plugin.

Note: I tried using the latest published version, `0.8.0`, but it failed with an unrelated error: `waiting for RPCs: marshaling properties: awaiting input property "spec": cannot marshal an input of type pulumi.String with element type string as a value of type pulumi.StringInput`.

I'm going to close this as a duplicate of https://github.com/pulumi/pulumi/issues/13223.
